installer/src/main/bin/shared.cmd

```diff
@@ -0,0 +1,250 @@
+    SHIFT
+    IF NOT "%~1" == "" (
+      GOTO arg_loop
+    )
+
+    @rem verify that ORACLE_HOME was set.
+    IF "%ORACLE_HOME%" == "" (
+      ECHO Required argument -oracle_home not provided >&2
+      EXIT /B 99
+    )
+GOTO :EOF
+
+:variableSetup
+    @REM set up variables for WLST or Jython execution
+
+    @REM set the WLSDEPLOY_HOME variable. if it was already set, verify that it is valid
+
+    IF NOT DEFINED WLSDEPLOY_HOME (
+      SET WLSDEPLOY_HOME=%SCRIPT_PATH%\..
+    ) ELSE (
+      IF NOT EXIST "%WLSDEPLOY_HOME%" (
+        ECHO Specified WLSDEPLOY_HOME of "%WLSDEPLOY_HOME%" does not exist >&2
+        EXIT /B 2
+      )
+    )
+    FOR %%i IN ("%WLSDEPLOY_HOME%") DO SET WLSDEPLOY_HOME=%%~fsi
+    IF %WLSDEPLOY_HOME:~-1%==\ SET WLSDEPLOY_HOME=%WLSDEPLOY_HOME:~0,-1%
+
+    @REM set up logger configuration, see WLSDeployLoggingConfig.java
+
+    SET LOG_CONFIG_CLASS=oracle.weblogic.deploy.logging.WLSDeployCustomizeLoggingConfig
+    SET WLSDEPLOY_LOG_HANDLER=oracle.weblogic.deploy.logging.SummaryHandler
+
+    IF NOT DEFINED WLSDEPLOY_LOG_PROPERTIES (
+      SET WLSDEPLOY_LOG_PROPERTIES=%WLSDEPLOY_HOME%\etc\logging.properties
+    )
+    IF NOT DEFINED WLSDEPLOY_LOG_DIRECTORY (
+      SET WLSDEPLOY_LOG_DIRECTORY=%WLSDEPLOY_HOME%\logs
+    )
+    IF NOT DEFINED WLSDEPLOY_LOG_HANDLERS (
+      SET WLSDEPLOY_LOG_HANDLERS=%WLSDEPLOY_LOG_HANDLER%
+    )
+GOTO :EOF
+
+:runJython
+    @REM run a jython script, without WLST.
+    SET JYTHON_SCRIPT=%1
+
+    @REM set up Oracle directory, logger, classpath
+
+    SET ORACLE_SERVER_DIR=
+    IF EXIST "%ORACLE_HOME%\wlserver_10.3" (
+        SET ORACLE_SERVER_DIR=%ORACLE_HOME%\wlserver_10.3"
+    ) ELSE IF EXIST "%ORACLE_HOME%\wlserver_12.1" (
+        SET ORACLE_SERVER_DIR=%ORACLE_HOME%\wlserver_12.1"
+    ) ELSE (
+        SET ORACLE_SERVER_DIR=%ORACLE_HOME%\wlserver
+    )
+
+    CALL :variableSetup
+    if %ERRORLEVEL% NEQ 0 (
+        EXIT /B %ERRORLEVEL%
+    )
+
+    SET "JAVA_PROPERTIES=-Djava.util.logging.config.class=%LOG_CONFIG_CLASS%"
+    SET "JAVA_PROPERTIES=%JAVA_PROPERTIES% -Dpython.cachedir.skip=true"
+    SET "JAVA_PROPERTIES=%JAVA_PROPERTIES% -Dpython.path=%ORACLE_SERVER_DIR%/common/wlst/modules/jython-modules.jar/Lib"
+    SET "JAVA_PROPERTIES=%JAVA_PROPERTIES% -Dpython.console="
+    SET "JAVA_PROPERTIES=%JAVA_PROPERTIES% %WLSDEPLOY_PROPERTIES%"
+
+    SET CLASSPATH=%WLSDEPLOY_HOME%\lib\weblogic-deploy-core.jar
+    SET CLASSPATH=%CLASSPATH%;%ORACLE_SERVER_DIR%\server\lib\weblogic.jar
+
+    @REM print the configuration, and run the script
+
+    ECHO JAVA_HOME = %JAVA_HOME%
+    ECHO CLASSPATH = %CLASSPATH%
+    ECHO JAVA_PROPERTIES = %JAVA_PROPERTIES%
+
+    SET PY_SCRIPTS_PATH=%WLSDEPLOY_HOME%\lib\python
+
+    ECHO ^
+    %JAVA_HOME%/bin/java -cp %CLASSPATH% ^
+        %JAVA_PROPERTIES% ^
+        org.python.util.jython ^
+        "%PY_SCRIPTS_PATH%\%JYTHON_SCRIPT%" %SCRIPT_ARGS%
+
+    %JAVA_HOME%/bin/java -cp %CLASSPATH% ^
+        %JAVA_PROPERTIES% ^
```

Comment:
The WLSDEPLOY_PROPERTIES is usually used to set WLST specific properties. I have always had a problem with the name of this property because I cannot think of a property that is specific to WDT that is set using this value. Unless its the typedef json file which isn't used in these scripts anyway. Opinion?

Reply:
WLSDEPLOY_PROPERTIES can be used by customer.  For example, setting -Doracle.jdbc.fanEnabled=false when creating a JRF domain against RAC.  WLST_PROPERTIES is reset in the script.  If we remove this, we need to document the changes but may break existing customer if they are using it.

Reply:
This shared script is for non-WLST scripts only, correct?

Reply:
This will eventually be the one for WLST scripts also.  It will re-use some of these methods.

Reply:
so back to the original comment. Do we need this property if we have JAVA_PROPERTIES? in the other scripts we set the WLST command argument to this value. In the encryptModel script (or the others that will use this shared script), you will never have the setting -Doracle.jdbc.fanEnabled=false

Reply:
To change the name of the property would break backward compatibility.  There needs to be a really REALLY good reason to do so.

Reply:
There is no backward compatibility. WLSDEPLOY_PROPERTIES was set on WLST_PROPERTIES (or whatever the wlst environment var is) and wlst does not exist in this new script.